ui/src/utils/QualityControl.js

```diff
@@ -40,11 +40,45 @@ export class QualityControl {
             errorMessage: '',
         };
 
+        const audioResult = this.soundCheck();
+
         if (this.audioBuffer.duration < 2.0) {
             qualityResult.success = false;
             qualityResult.errorMessage += 'Audio recording failed: recording was too short. Try again';
+        } else if (audioResult) {
+            qualityResult.success = false;
+            qualityResult.errorMessage += audioResult;
         }
 
         return qualityResult;
     }
+
+   /**
+    * Checks if the sound is too silent
+    */
+    soundCheck() {
+        const bufferArray = this.audioBuffer.getChannelData(0);
+        const bufferSet = new Set(bufferArray);
+        const arrOfSet = [];
+        for (let data of bufferSet) {
+            arrOfSet.push(data);
+        }
+        const arrOfBiggest = [];
```

Comment:
to find the n largest items in an array, you can sort it then take the elements off the end

```javascript
function nLargest(arr, n) {
  // copy it because sort mutates the array
  // sort is ascending order, so reverse it, now largest elements are in front
  const sorted = [...arr].sort().reverse();
  return sorted.slice(n); // returns array of first n in sorted
}
```